app/src/main/java/org/mozilla/fenix/components/metrics/AppLaunchTimeMeasurement.kt

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.metrics
+
+import android.os.Process
+import android.os.SystemClock
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.MainScope
+import kotlinx.coroutines.launch
+import org.mozilla.fenix.perf.Stat
+
+/**
+ * Handles the logic of figuring out launch time for cold, warm and hot startup.
+ */
+class AppLaunchTimeMeasurement(private val stats: Stat = Stat()) {
+
+    // when user launched the time. i.e press app_icon, launched a link.
+    private var processLaunchTimeStampNanoSeconds: Long? = null
+    // we are considering screen to be visible after the first pre draw call.
+    private var elapsedNanoTimestampOnHomeActivityPreDraw: Long? = null
+
+    fun onApplicationOnCreate() =
+        MainScope().launch(Dispatchers.IO) {
+        processLaunchTimeStampNanoSeconds = stats.getProcessStartTimeStampNano(Process.myPid())
```

Comment:
There is a race condition: this may not be set by the time we record the metric (and confusingly `ActivityonCreate` though the end result should be the same).

Reply:
hmm would it make sense to keep track of whether or not the application oncreate was called. and than during onPreDraw() launch a coroutine to collect this data?

Reply:
Yeah, I think so – fetch it just prior to submitting.